src/Analysis/Ast/Impl/Specializations/BuiltinsSpecializations.cs

```diff
@@ -91,6 +93,73 @@ public static IMember List(IPythonInterpreter interpreter, IPythonFunctionOverlo
             return args.Count > 0 && args[0] is PythonCollection c ? c.Contents.FirstOrDefault() : null;
         }
 
+        public static IMember Super(IPythonModule declaringModule, IPythonFunctionOverload overload, IArgumentSet argSet, IndexSpan indexSpan) {
+            var args = argSet.Values<IMember>();
+
+            if (args.Count == 0) {
+                //Zero argument form only works inside a class definition
+                foreach (var s in argSet.Eval.CurrentScope.EnumerateTowardsGlobal.Where(s => s.Node is ClassDefinition)) {
+                    var classType = s.Variables["__class__"].GetPythonType<IPythonClassType>();
+                    return CreateSuper(argSet, classType, declaringModule, indexSpan);
+                }
+                return null;
+            }
+
+            // If multiple arguments first argument is required
+            var firstCls = args.FirstOrDefault().GetPythonType<IPythonClassType>();
+            if (firstCls == null) {
+                return null;
+            }
+
+            // second argument optional
+            bool isUnbound = args.Count == 1;
+            if (isUnbound) {
+                return CreateSuper(argSet, firstCls, declaringModule, indexSpan);
+            }
+
+            var secondCls = args[1].GetPythonType<IPythonClassType>();
+            if (secondCls?.Equals(firstCls) == true || 
+                secondCls?.IsSubClassOf(firstCls) == true) {
+                // We walk the mro of the second parameter looking for the first
+                return CreateSuper(argSet, secondCls, declaringModule, indexSpan, typeToFind: firstCls);
+            }
+
+            return null;
+        }
+
+        private static IMember CreateSuper(IArgumentSet argSet,
+            IPythonClassType classType,
+            IPythonModule declaringModule,
+            IndexSpan indexSpan,
+            IPythonType typeToFind = null) {
+
+            var mro = classType?.Mro ?? Array.Empty<IPythonType>();
+            if (mro.Count == 0) {
```

Comment:
I moved the null checking from outside of CreateSuper, to here, tried to combine null checking with count == 0 checking. came up with this